elastic/beats
# Add username/password in Metricbeat autodiscover hints

This PR closes https://github.com/elastic/beats/issues/15115

### Manual Testing
1. Enable autodiscover in metricbeat.yml
```
metricbeat:
  autodiscover.providers:
    - type: docker
      hints.enabled: true
```
2. Start metricbeat with log enabled: `./metricbeat -e -d "hints.builder" `
3. `docker run -l co.elastic.metrics/module=nats -l co.elastic.metrics/username=user42 --name nats nats`
4. Check in Metricbeat logs for : `2019-12-16T15:53:18.712+0200	DEBUG	[hints.builder]	hints/metrics.go:144	generated config: {"enabled":true,"hosts":null,"metricsets":["connections","routes","stats","subscriptions"],"module":"nats","period":"1m","processors":null,"ssl":null,"timeout":"3s","username":"user42"}`
